libsrc/osic1p/cputc.s

```diff
@@ -48,8 +48,23 @@ newline:
         lda     CURS_Y
         cmp     #SCR_HEIGHT     ; screen height
         bne     plot
-        lda     #0              ; wrap around to line 0
-        sta     CURS_Y
+        dec     CURS_Y          ; bottom of screen reached, scroll
+        ldx     #0
+scroll: lda     SCRNBASE+$00A5,x
+        sta     SCRNBASE+$0085,x
+        lda     SCRNBASE+$01A5,x
+        sta     SCRNBASE+$0185,x
+        lda     SCRNBASE+$02A5,x
+        sta     SCRNBASE+$0285,x
```

Comment:
I don't know the target well but in general I'd presume that these "magic constants" could be replaced by some simple expressions (beside 'plus' you have i.e. 'multiply' and 'shift') and potentially using binary constants instead of hex constants that makes to more obvious what they mean.